include/dlaf/common/callable_object.h

```diff
@@ -21,12 +21,12 @@
 /// The function name is wrapped in parentheses to disable ADL. It is assumed
 /// that all required overloads are found relative to the namespace where the
 /// macro is used.
-#define DLAF_MAKE_CALLABLE_OBJECT(fname)                                                   \
-  constexpr struct fname##_t {                                                             \
-    template <typename... Ts>                                                              \
-    auto operator()(Ts&&... ts) const noexcept(noexcept((fname)(std::forward<Ts>(ts)...))) \
-        -> decltype((fname)(std::forward<Ts>(ts)...)) {                                    \
-      return (fname)(std::forward<Ts>(ts)...);                                             \
-    }                                                                                      \
-  } fname##_o {                                                                            \
+#define DLAF_MAKE_CALLABLE_OBJECT(fname)                                                    \
+  constexpr struct fname##_t {                                                              \
+    template <typename... Ts>                                                               \
+    auto operator()(Ts&&... ts) const noexcept(noexcept((fname) (std::forward<Ts>(ts)...))) \
+        -> decltype((fname) (std::forward<Ts>(ts)...)) {                                    \
```

Comment:
Probably you might want to check this
```suggestion
        -> decltype((fname)(std::forward<Ts>(ts)...)) {                                    \
```